Pipeline.py

```diff
@@ -1,17 +1,27 @@
 import sys
 
-from pdart.Logging import init_logging
+from pdart.Logging import init_logging, PDS_LOGGER
 from pdart.pipeline.Directories import Directories, make_directories
 from pdart.pipeline.StateMachine import StateMachine
 
+# -1: print out every message
+INFO_MESSAGE_LIMIT = -1
+
 
 def run() -> None:
     assert len(sys.argv) == 2, sys.argv
     proposal_id = int(sys.argv[1])
     init_logging()
     dirs = make_directories()
+    PDS_LOGGER.open(
+        f"Pipeline for proposal id: {proposal_id}",
+        limits={
+            "info": INFO_MESSAGE_LIMIT,
+        },
+    )
     state_machine = StateMachine(dirs, proposal_id)
     state_machine.run()
+    PDS_LOGGER.close()
 
 
 if __name__ == "__main__":
```

Comment:
This looks really good. When you open a logger, as with opening a file, you should take care to be 100% certain the logger gets closed, even in the event of an exception. So I recommend you put "try:" immediately after PDS_LOGGER.open, and put "finally:" before PDS_LOGGER.close(). Also, PdsLogger has a handy method to put an exception into the log file. Before the "finally:", add `except Exception as e: PDS_LOGGER.error(e)`. You will get the error message and stacktrace in the log file, so you can see exactly what went wrong.